### PR TITLE
Improvements to move page feature

### PR DESCRIPTION
**Pull request by WickyNilliams**
### What is the context of this PR?
* Simplifies some awkward logic around computing the `position` value for pages. Possibly needs some indexes or something added. Need to investigate further. But for now the code works
* Simplifies logic around calculating `order` values when moving pages
* Reduces number of round trips to DB, at cost of bigger payload from DB
* Improves performance of inserts - previously would insert at end of list, _then_ perform a move operation to place at correct position. Now inserts directly at correct position

### How to review 
Run tests. Run e2e tests. Check code is sensible

### CODE REVIEW

File: repositories/PageRepository.js [hunk 1]
```diff
@@ -68,33 +62,42 @@ module.exports.insert = function insert(args) {
...
+Object.assign(module.exports, {
+  findAll,
+  get: getById,
+  insert,
+  update,
+  remove,
+  undelete,
+  move,
+  getPosition
+});
```
Comment (samiwel) [inline repositories/PageRepository.js]: Nice tidy up 👍

Comment (WickyNilliams) [reply repositories/PageRepository.js]: It's a clever trick I saw here: https://twitter.com/getify/status/991427888359596032.

It saves on repetitious `module.export.whatever`, and prevents circular dependencies from causing issues.

**Comment by WickyNilliams** [comment]
[Consider using View table to simplify data access logic around reordering](https://trello.com/c/HePJ415f/423-consider-using-view-table-to-simplify-data-access-logic-around-reordering)

File: repositories/strategies/spacedOrderStrategy.js [hunk 2]
```diff
@@ -1,104 +1,54 @@
...
+
+  return isNil(id) ? query : query.where("id", "!=", id);
 };
 
-const movePage = async (trx, { id, sectionId, position }) => {
-  position = Math.max(0, position);
+const getMaxOrder = flow(last, getOr(0, "order"));
 
-  let rows = await getAdjacentRows(trx, id, sectionId, position);
+const makeSpaceForInsert = async (trx, sectionId, id, position) => {
```
Comment (WickyNilliams) [inline repositories/strategies/spacedOrderStrategy.js]: Not sure what to call this function. This doesn't seem like a great name, as it doesn't always "make space". Sometimes it's just doing a calculation, and others it's updating rows in the DB.

Anyone have any thoughts on this?

Comment (SamGodwin2) [reply repositories/strategies/spacedOrderStrategy.js]: Could (get/find)OrderForInsert() work?

**Pull request merged**